### ISSUE TITLE
[Bug] There are some difficulties in clicking.

### ISSUE THREAD

**Issue by boludoz** [kaliiiiiiiiii/Selenium-Driverless#192]
The clicking system is great, but sometimes it is not correct when there is movement, the button cannot be pressed and it causes a silent error.

**Comment by kaliiiiiiiiii** [comment]
@boludoz Please provide more information such as
- movement of what? 
- what's the `silent error`?
- Code to test with (POC)

**Comment by boludoz** [comment]
For example, suppose we disable the loading wait in 'self.get' and use the element wait timeout, which is very useful.

If the website does not finish loading and an image appears that moves a button downwards in a cascading manner, this will cause a click failure, and it is more common than it seems.

**Comment by kaliiiiiiiiii** [comment]
You mean support `moving elements` ?

**Comment by boludoz** [comment]
Yes, I think it is more of a "small bug", if an element is not loaded on time it will fail, using javascript to click that does not happen.

**Comment by boludoz** [comment]
Maybe you can make the click location check and the click be in the same previous call

**Comment by kaliiiiiiiiii** [comment]
> Maybe you can make the click location check and the click be in the same previous call

Nope, click unfortunately has to be in one call. Only thing possible similarly like that is pausing the debugger. Tho thet ofc is detectable.

**Comment by boludoz** [comment]
> > Maybe you can make the click location check and the click be in the same previous call
> 
> Nope, click unfortunately has to be in one call. Only thing possible similarly like that is pausing the debugger. Tho thet ofc is detectable.

Check repeatedly that the button does not change position.

**Comment by boludoz** [comment]
Maybe a pre-check.

**Comment by boludoz** [comment]
Perhaps a pre-check of the element as a human would do, pausing the page when clicking.

**Comment by boludoz** [comment]
Can you explain to me how you did this to access an iframe with CDP?

**Comment by kaliiiiiiiiii** [comment]
> Can you explain to me how you did this to access an iframe with CDP?

business secret 🤫😂
Nah it's in the code ofc but - to much to explain for me:/

**Comment by boludoz** [comment]
Perhaps you can pause the execution of the website for a moment.

**Comment by kaliiiiiiiiii** [comment]
> Perhaps you can pause the execution of the website for a moment.

yeah but that would be detectable

**Comment by boludoz** [comment]
> > Perhaps you can pause the execution of the website for a moment.
> 
> yeah but that would be detectable

Can't you cut off the internet and disable the cache just before clicking? I think there is a command to do this.

**Comment by kaliiiiiiiiii** [comment]
> > > Perhaps you can pause the execution of the website for a moment.
> > 
> > 
> > yeah but that would be detectable
> 
> Can't you cut off the internet and disable the cache just before clicking? I think there is a command to do this.

uh well but the script could save the current time in a variable, even obfuscated etc. just really not feasible. 
Or detect timing lags over a websocket at server side.

This will have to get implemented by some propper algorithm which takes into account moving elements & generates a mouse path accordingly in real-time.

atm don't have the time nor enough calculus & stats. knowledge to develop that kind of stuff.

**Comment by boludoz** [comment]
> > > > Perhaps you can pause the execution of the website for a moment.
> > > 
> > > 
> > > yeah but that would be detectable
> > 
> > 
> > Can't you cut off the internet and disable the cache just before clicking? I think there is a command to do this.
> 
> uh well but the script could save the current time in a variable, even obfuscated etc. just really not feasible. Or detect timing lags over a websocket at server side.
> 
> This will have to get implemented by some propper algorithm which takes into account moving elements & generates a mouse path accordingly in real-time.
> 
> atm don't have the time nor enough calculus & stats. knowledge to develop that kind of stuff.

It is not a moving element, but a simple web page that has not yet finished loading its images and there is a button right below it.

**Comment by kaliiiiiiiiii** [comment]
> It is not a moving element, but a simple web page that has not yet finished loading its images and there is a button right below it.

Well but it's still the issue that the element location changes - right?
Soo therefore still same approach

**Comment by boludoz** [comment]
It's true
